.github/workflows/ci.yml

```diff
@@ -97,6 +97,9 @@ jobs:
           # The main Error Prone run
           $MAVEN ${MAVEN_TEST} -T C1 clean test-compile -P gib,errorprone-compiler \
             -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
+          # Ensure `verify` works with error-prone compiler, since it's what developers may use locally
+          $MAVEN ${MAVEN_TEST} -T C1 clean test-compile -P gib,verify \
```

Comment:
Where is this `verify` profile defined?

Reply:
Can't this be enabled in the previous command?

Reply:
> Where is this `verify` profile defined?

it's not a profile. it's a "target"?
something built-in in maven.

my bad, fixing

> Can't this be enabled in the previous command?

hm, maybe.

Reply:
I'm confused. `verify` runs integration tests, and the point of this job is to only compile the project. If we want to run tests with error-prone enabled, we should merge this job with others.

Reply:
OK, you know more about mvn than i do. I don't know what are the integration tests, but it's not my intention here.

This is the command i run quite often:

```
mvnd verify -DskipTests -Dmaven.javadoc.skip=true -Dmaven.source.skip=true -Dskip.npm -Dskip.yarn -nsu -P errorprone-compiler -pl '!:trino-server,!:trino-server-rpm,!:trino-docs'
```

it recently got broken (by https://github.com/trinodb/trino/pull/13108) and i wanted to prevent this from being the case in the future.

while this may not be the right command, the breakage was real (maven plugin verifying dependencies would fail).
obviously `test-compile` target didn't run it implicitly, while `verify` did

this is now fixed (https://github.com/trinodb/trino/pull/13227), so i am fine parking this for now.

Reply:
It's a [lifecycle phase](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#default-lifecycle), not a "target". The phases run in order, so `verify` will include the `test-compile` phase. Each phase includes all previous phases in the given lifecycle.

Reply:
@electrum thanks for clarifying. Do you know how to run all the static code analysis (checkstyle, error-prone, maven dependency check, etc), but without running tests nor integration tests?

Reply:
`verify` is the phase just before `install` so generally that's what all projects use to run all checks that exist in the project. I agree with Piotr that we should fix this regardless of what CI uses or does not use.

Reply:
I suggest enabling error prone in the `Maven check` jobs. The added overhead might be smaller than compiling the whole project again in this job.

Reply:
error-prone isn't our main compiler. we need to ensure verify works and static code analysis passes also when error-prone isn't used
i assume error-prone _may_ have a bug, see error-prone -specific workaround here https://github.com/trinodb/trino/blob/0cba6420b7a0b47a96ee8f53505edd80a052696b/plugin/trino-phoenix5/pom.xml#L392-L403

Reply:
So we could run `Maven check` twice, with and without error prone. It could still be a single job with a matrix. There's already a matrix that we used for different Java versions, we could add profiles there.

Reply:
Doing in single job makes it a bit more work to know what failed. If I see maven-checks pass but errorprone fail I know what to run locally.

Reply:
Proper usage of annotations (that is without false positives) should help.